api/client/proto/authservice.proto

```diff
@@ -804,6 +804,42 @@ message SingleUseUserCert {
     }
 }
 
+message GetEventsRequest {
+    // Namespace, if not set, defaults to 'default'
+    string Namespace = 1;
+    // Oldest date of returned events
+    google.protobuf.Timestamp StartDate = 2
+        [ (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
+    // Newest date of returned events
+    google.protobuf.Timestamp EndDate = 3
+        [ (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
+    // EventTypes is optional, if not set, returns all events
+    repeated string EventTypes = 4;
+    // Maximum amount of events returned
+    int32 Limit = 5;
+    // When supplied the search will resume from the last key
+    string StartKey = 6;
+}
+
+message GetSessionEventsRequest {
+    // Oldest date of returned events
+    google.protobuf.Timestamp StartDate = 1
+        [ (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
+    // Newest date of returned events
+    google.protobuf.Timestamp EndDate = 2
+        [ (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
+    int32 Limit = 3;
+    string StartKey = 4;
+}
+
+message Events {
+    bytes Items = 1;
```

Comment:
This should be a `repeated events.OneOf`.

Reply:
See other comment.